contracts/modules/token/TokenController.sol

```diff
@@ -189,36 +199,48 @@ contract TokenController is ITokenController, LockHandler, MasterAwareV2 {
     return token.totalSupply();
   }
 
-  /// Returns the base voting power not the balance. It is used in governance voting as well as in
-  /// snapshot voting.
+  /// Returns the base voting power. It is used in governance and snapshot voting.
+  /// Includes the delegated tokens via staking pools.
   ///
-  /// @dev Caution, this function is improperly named because reconfiguring snapshot voting was
-  /// not desired. It accounts for the tokens in the user's wallet as well as tokens locked in
-  /// assessment and legacy staking deposits. V2 staking deposits are excluded because they are
-  /// delegated to the pool managers instead.
-  /// TODO: add stake pool balance for pool operators
+  /// @param _of  The member address for which the base voting power is calculated.
+  function totalBalanceOf(address _of) public override view returns (uint) {
+    return _totalBalanceOf(_of, true);
+  }
+
+  /// Returns the base voting power. It is used in governance and snapshot voting.
+  /// Does not include the delegated tokens via staking pools in order to act as a fallback if
+  /// voting including delegations fails for whatever reason.
   ///
   /// @param _of  The member address for which the base voting power is calculated.
-  function totalBalanceOf(address _of) public override view returns (uint256 amount) {
+  function totalBalanceOfWithoutDelegations(address _of) public override view returns (uint) {
+    return _totalBalanceOf(_of, false);
+  }
 
-    amount = token.balanceOf(_of);
+  function _totalBalanceOf(address _of, bool includeManagedStakingPools) internal view returns (uint) {
+
+    uint amount = token.balanceOf(_of);
 
     // This loop can be removed once all cover notes are withdrawn
     for (uint256 i = 0; i < lockReason[_of].length; i++) {
       amount = amount + _tokensLocked(_of, lockReason[_of][i]);
     }
 
-    // [todo] Can be removed after PooledStaking is decommissioned
-    uint stakerReward = pooledStaking().stakerReward(_of);
-    uint stakerDeposit = pooledStaking().stakerDeposit(_of);
+    // TODO: can be removed after PooledStaking is decommissioned
+    amount += pooledStaking().stakerReward(_of);
+    amount += pooledStaking().stakerDeposit(_of);
+
+    (uint assessmentStake,,) = assessment().stakeOf(_of);
+    amount += assessmentStake;
 
-    (
-      uint assessmentStake,
-      /*uint104 rewardsWithdrawableFromIndex*/,
-      /*uint16 fraudCount*/
-    ) = assessment().stakeOf(_of);
+    if (includeManagedStakingPools) {
+      uint managedStakingPoolCount = managerStakingPools[_of].length;
+      for (uint i = 0; i < managedStakingPoolCount; i++) {
+        uint poolId = managerStakingPools[_of][i];
+        amount += stakingPoolNXMBalances[poolId].deposits;
```

Comment:
how about the .rewards why not add those too here?

Reply:
Technically claims and governance rewards were never accounted for here, btw, main reason being gas costs. PooledStaking rewards were - but not in a delegation context, just your rewards. From the discussion it seems to be preferred to not delegate the rewards - so we'll leave that one out. Accounting for your own staking rewards in `totalBalanceOf` would be costly, so we're leaving that one out as well - it will be consistent across the system not to include rewards in the `totalBalanceOf`.

Reply:
Understood, here it would be cheap to add obviously but for consistency, it would need to always use rewards to be delegated, that makes sense.